### DISCUSSION TITLE
Add Support for Heltec BMS (= YYBMS YanYang)

### DISCUSSION THREAD

**Discussion by ramack** [Louisvdw/dbus-serialbattery#575] [New BMS requests]
**Is your feature request related to a problem? Please describe.**
I read on https://github.com/Louisvdw/dbus-serialbattery/wiki/BMS-types-supported that "JKBMS / Heltec BMS" are supported, so I tried to connect my Heltec Smart BMS 9-25S 100A RS485 but it is not detected by the dbus-serialbattery driver.

**Describe the solution you'd like**
According the docs I have (and what I spied on the RS485 with the scope) the BMS seems to talk Modbus RTU and is different from the JKBMS. So I propose to add support for this Heltec device. I don't know about other Heltec BMSs and if there is a variant that is compatible to JKBMS, but we might also want to update the docs accordingly.

**Comment by ramack** [comment]
I am working on this. @Louisvdw  @mr-manuel should I base that work directly on the JKBMS-BLE branch?

**Comment by mr-manuel** [comment]
If you are not adding a BLE driver than you can merge with the master branch. The only thing is that there are a lot of changes outstanding, see https://github.com/Louisvdw/dbus-serialbattery/pull/468.
If you are adding only a new python file and the corresponding additions to include the file it should not be hard to merge.

**Comment by ramack** [comment]
OK thanks. The config.ini seems to be a very interesting change there... maybe the timeline to merge that into master us the main question right now

**Comment by mr-manuel** [comment]
Unfortunately @Louisvdw has not much time right now, but he will do so after 2nd May. If you want to keep it simple merge with the `jkbms_ble` branch, than I can review your PR and merge it directly.

**Comment by ramack** [comment]
Out of curiosity: what is the difference between an issue and a discussion?

BTW I have the driver in production already, but it will take some time to cleanup until I can file a PR.

**Reply by mr-manuel** [reply]
In a discussion you can comment answers and is thought for discussing topics. Issues are thought for issues, bugs, support and festure requests :)

**Comment by ramack** [comment]
The protocol is here: [HeltecBMS_smrt_bms,modbus_protocol_v1.5.pdf](https://github.com/Louisvdw/dbus-serialbattery/files/11536852/HeltecBMS_smrt_bms.modbus_protocol_v1.5.pdf)

**Comment by mr-manuel** [comment]
This is implemented with https://github.com/Louisvdw/dbus-serialbattery/issues/508
